sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java

```diff
@@ -114,7 +114,7 @@
    * @param <T> the element type
    */
   public static <T> AvroCoder<T> of(Class<T> clazz) {
-    return new AvroCoder<>(clazz, ReflectData.get().getSchema(clazz));
+    return new AvroCoder<>(clazz, new ReflectData(clazz.getClassLoader()).getSchema(clazz));
```

Comment:
In this case I don't think it will break, but doesn't it need to be the classloader of `clazz`? Here I don't think it matters, but elsewhere the idea is that `clazz` might get stuck in a cache of the `ReflectData`.